pymc3/sampling.py

```diff
@@ -562,14 +593,27 @@ def sample(
     )
 
     if compute_convergence_checks:
+        # convert trace to InferenceData, with awareness of warmup!
+        # arviz 0.7.0 doesn't ignore warmup draws if they are in the trace! (see arviz #1146)
+        # that's why here, we make sure to only put actual posterior samples into idata.posterior
+        trace_warmup = trace[:-n_draws]   # <-- may result in len(trace_warmup) == 0
+        trace_posterior = trace[-n_draws:]
+        idata = arviz.from_pymc3(trace_posterior, log_likelihood=False, model=model)
+        # save additional sampling metadata
+        idata.posterior.attrs['n_tune'] = n_tune
+        idata.posterior.attrs['n_draws'] = n_draws
+        idata.posterior.attrs['t_sampling'] = t_sampling
```

Comment:
Isn't all that done under the hood by `arviz.from_pymc3` now (cf Oriol's PR)?

Reply:
Yes it is. Same reason as above.. I'm looking forward to take it out.

Reply:
Indeed, this will be solved if we pin latest ArviZ 👌

Reply:
The version is now pinned and the code got much simpler. Please check the latest changes.